site/en/dev-best-practice.md

```diff
@@ -0,0 +1,145 @@
+
+  * Increase `MAJOR` by one if a changeset introduces **breaking changes**.
+  * Increase `MINOR` by one if a changeset introduces **new features**.
+  * Increase `PATCH` by one if a changeset **fixes problems with the minor version**.
+
+You should start versioning at `0.0.1`. While `MAJOR` is `0`, the software is to be considered alpha but you should still stick to the scheme for minor and patch revisions. Once you consider the software mature enought to merit a `1.0.0` release, take extra care to adhere to the convention as much as possible.
+
+The advantage of semantic versioning is that it conveys the consequences of an upgrade to the user. There is never a reason not to upgrade to the latest patch release whereas a upgrading to a new major version should be undertaken with care.
+
+## Licensing
+
+OCR-D is firmly committed to the values of [F(L)OSS](https://en.wikipedia.org/wiki/Free_and_open-source_software) and [permissive licensing](https://en.wikipedia.org/wiki/Permissive_software_license), as these not only encourage community use and uptake, but also create trust between all stakeholders and transparency of the development process.
+
+OCR-D favours the [Apache Software License 2.0](https://www.apache.org/licenses/LICENSE-2.0) (ASL) for it is a permissive license that supports the waiving of liabilities while still granting sufficient freedom for e.g. commercial parties to use and redistribute products that include source code from OCR-D.
+
+Second to the ASL, other permissive open source licenses that can be used include (in order of preference): [MIT](https://en.wikipedia.org/wiki/MIT_License), [BSD](https://en.wikipedia.org/wiki/BSD_licenses). If there are strong reasons that prevent using a permissive license for your project, please [get in touch with the coordination project](https://ocr-d.de/en/contact) to discuss other licensing options. <!--Weakly permissive licenses such as e.g. GPL, LGPL, APGL should only be used when there are strong reasons against the use of a permissive license. -->
+
+## Python project setup
+
+Since most processors are implemented in Python, we have been developing certain conventions that have proven effective for interoperability, deployment and robustness. Following these conventions makes it easier to contribute and frees developers from the grunt work of managing a project, to spend more time on development, documentation and support.
+
+Have a look at existing projects that follow these cvonventions, e.g. [`ocrd_tesserocr`](https://github.com/OCR-D/ocrd_tesserocr). Much can be copied, pasted and adapted.
+
+### Project layout
+
+When starting a new project `ocrd_foo`, create this folder layout:
+
+```sh
+ocrd_foo/
+ocrd_foo/__init__.py
+tests/test_foo.py
+LICENSE
+README.md
+setup.py
+```
+
+### `setup.py`
+
+We use `setuptools` for python packaging in OCR-D.
+
+### `tests/`
+
+### `LICENSE`
+
+## Deployment
```

Comment:
should point to:
- our PYTHON/VIRTUAL_ENV customization
- our target conventions (`deps-ubuntu`, `install`, `test`)
- the GNU make texinfo manual for reference
```suggestion
## Makefile

## Deployment
```